common/remote/sockfile.cpp

```diff
@@ -5745,13 +5745,29 @@ class CRemoteFileServer : implements IRemoteFileServer, public CInterface
             }
         }
 
-        if (_useSSL)
+        if (_connectMethod == SSLOnly || _connectMethod == SSLFirst || _connectMethod == UnsecureFirst)
         {
             if (sslep.port == 0)
-                throw createDafsException(DAFSERR_connection_failed, "Secure dafilesrv port not specified");
+                throw createDafsException(DAFSERR_serverinit_failed, "Secure dafilesrv port not specified");
 
-            if ( (_useSSL != UnsecureFirst) && (!securitySettings.certificate || !securitySettings.privateKey) )
-                throw createDafsException(DAFSERR_connection_failed, "SSL Certificate and/or Key file information not found in environment.conf");
+            if (_connectMethod == UnsecureFirst)
```

Comment:
I am concerned that if the admin misconfigures, specifying UnsecureFirst but forgot to specify the certificates, you will still start up but not listen on the secure port. SSL clients will not be able to connect, and there is no real way for someone browsing logfiles to know why.  I believe startup should fail with a helpful error message